### ISSUE TITLE
Provide more visibility for server errors and improve logging

### ISSUE THREAD

**Issue by jrozner** [theparanoids/ashirt-server#30]
Currently it's pretty difficult to identify where an issue is occurring in the case of a server failure. Any condition that is going to result in a 5xx error should provide a stack trace. We should put some thought into logging overall and and ensure that it is both easy to ingest, easy to read, and easy to act on. Current logging is alright and at least traceable due to the request identifiers. Shoving a stack trace into the log event may not be as trivial due to needing to handle escaping the newlines and quotes for the the structured logging used. Let's spend some time reviewing what fields are present and exactly how to handle this.

**Comment by JoelAtDeluxe** [comment]
It looks like there are a few libraries we might be able to use to produce a stack trace for us, but I need to investigate them.

I'm wondering if a stacktrace is the proper solution. Do we really need to know specifically what line failed in most cases? I think we may want to just start using wrapped errors more. That should help us pinpoint the error just by searching for the error text. Just as a quick note, a wrapped error would look like this, codewise:

```go
// this is a bit of psuedo code -- I don't remember the actual interface here
func outer() {
  err := inner()
  if err != nil {
    errBuff := ""
    for err.IsWrapped {
      errBuff = err.Error() + " >> " + errBuff
      err = err.Unwrap()
    }
   errBuff = err.Error() + errBuff;
   fmt.Printf("Got an error: %v\n", errBuff)
  }
  fmt.Println("Success!")
}
func inner() {
  err := someFunc()
  if err != nil {
    return error.Wrap("Inner got an error", err) // each situation where we needed to return an error would do this: wrap the error received with a new message, and include the original error
  }
}

func someFunc() {
  return fmt.Errorf("oops!")
}
```

The result of that would look like this: "Got an error: Inner  got an error >> oops", resulting an effective stacktrace, but losing specific line numbers. 

As for escaping newlines and quotes: I believe kitlog will do the quotes for us, but I'll test that. Newlines we would need to worry about, but those should be easy to remove, and unlikely to occur if we use error messages, rather than a stacktrace

Note that this approach, and I imagine any other approach, is likely going to affect the whole backend. Every error is likely to be touched here. We want to do this while we aren't waiting on any other PRs

**Comment by jrozner** [comment]
We shouldn't need any additional libraries to produce them. debug.Stack() is available in std which returns the current stack trace of, I believe, all the threads. At a previous company where we built a web framework and wanted similar behavior this is how we achieved it. I'm open to the idea of additional wrapping as an alternative assuming it provides sufficient visibility. Is it possible to see an example log event of where you're expecting to get to with wrapped events before deciding?

Realistically I think I'd only expect to see stack traces when a 5xx error occurs which suggests that it is a server (programming) error. Having line visibility is nice because it makes identifying where the error occurred fairly straight forward.
